src/main/java/com/shop/service/dataservice/product/ProductServiceImpl.java

```diff
@@ -172,59 +167,66 @@ public ProductWithDetailDto getCarWithDetail(long id) throws ProductNotFoundExce
 
         //TODO
         Pageable limit = new PageRequest(0, 100);
-        return carRepository.findAllByUserIdAndDeletedIsFalse(userId, limit);
+        return productRepository.findAllByUserIdAndDeletedIsFalse(userId, limit);
     }
 
     @Override
     @Transactional(readOnly = true)
     public ProductEntity getByIdAndUserId(Long id, Long userId) throws InvalidProductOwnerException {
-        ProductEntity carEntity = carRepository.findByIdAndUserIdAndDeletedIsFalse(id, userId);
-        if (carEntity == null)
-            throw new InvalidProductOwnerException(String.format("User with id %s can't edit car with id %s", userId, id));
+        ProductEntity productEntity = productRepository.findByIdAndUserIdAndDeletedIsFalse(id, userId);
+        if (productEntity == null)
```

Comment:
not if (productEntity == null)
use findByIdAndDeletedIsFalse, after that check owner

need something like  productEntity.getOwner().getId().equals(userId)